src/cargo.rs

```diff
@@ -0,0 +1,115 @@
+use std::ffi;
+use std::process;
+use std::str;
+use std::vec;
+
+use failure;
+use serde;
+use serde_json;
+
+const CURRENT_TARGET: &str = include_str!(concat!(env!("OUT_DIR"), "/current_target.txt"));
+
+#[derive(Debug)]
+pub struct Cargo {
```

Comment:
This new file is generating a lot of warnings from unused code.

I developed more than I'd need.  I plan to eventually split this out into is own crate.  It'll be reused by `stager` and other tools I'm working on that need to run `cargo` programmatically but don't want to link against `cargo` (slow to build, .  This was a testing ground for the idea.